packages/backend/src/modules/tracked-txs/modules/liveness/indexers/AnomaliesIndexer.test.ts

```diff
@@ -0,0 +1,331 @@
+import { ProjectId, UnixTime } from '@l2beat/shared-pure'
+import { expect, mockFn, mockObject } from 'earl'
+import { Project } from '../../../../../model/Project'
+import { IndexerService } from '../../../../../tools/uif/IndexerService'
+import { SavedConfiguration } from '../../../../../tools/uif/multi/types'
+import {
+  AnomaliesRecord,
+  AnomaliesRepository,
+} from '../repositories/AnomaliesRepository'
+import {
+  LivenessRecordWithSubtype,
+  LivenessRepository,
+} from '../repositories/LivenessRepository'
+import {
+  AnomaliesIndexer,
+  AnomaliesIndexerIndexerDeps,
+} from './AnomaliesIndexer'
+
+const NOW = UnixTime.now()
+const MIN = NOW.add(-100, 'days')
+
+const MOCK_CONFIGURATION_ID = createTrackedTxId.random()
+
+const MOCK_PROJECTS = [
+  mockObject<Project>({
+    projectId: ProjectId('mocked-project'),
+    isArchived: false,
+    trackedTxsConfig: [
+      mockObject<TrackedTxConfigEntry>({
+        id: MOCK_CONFIGURATION_ID,
+        type: 'liveness',
+        subtype: 'batchSubmissions',
+        untilTimestamp: UnixTime.now(),
+      }),
+    ],
+  }),
+]
+
+const MOCK_CONFIGURATIONS = [
+  mockObject<Omit<SavedConfiguration<TrackedTxConfigEntry>, 'properties'>>({
+    id: MOCK_CONFIGURATION_ID,
+    maxHeight: null,
+    currentHeight: 1,
+  }),
+]
+
+describe(AnomaliesIndexer.name, () => {
+  describe(AnomaliesIndexer.prototype.update.name, () => {
+    it('should skip update', async () => {
+      const indexer = createIndexer({
+        tag: 'update-skip',
+      })
+
+      const mockcalculateAnomalies = mockFn().resolvesTo([])
+      indexer.getAnomalies = mockcalculateAnomalies
+
+      const from = MIN.toNumber()
+      const to = NOW.add(-2, 'days').toNumber()
+
+      const result = await indexer.update(from, to)
+
+      expect(mockcalculateAnomalies).not.toHaveBeenCalled()
+
+      expect(result).toEqual(to)
+    })
+
+    it('should adjust and update', async () => {
+      const mockAnomaliesRepository = mockObject<AnomaliesRepository>({
+        addOrUpdateMany: mockFn().resolvesTo(1),
+      })
+
+      const indexer = createIndexer({
+        tag: 'update',
+        anomaliesRepository: mockAnomaliesRepository,
+      })
+
+      const mockAnomalies: AnomaliesRecord[] = [
+        {
+          timestamp: NOW.add(-1, 'days'),
+          projectId: MOCK_PROJECTS[0].projectId,
+          subtype: 'batchSubmissions',
+          duration: 100,
+        },
+      ]
+
+      const mockCalculateAnomalies = mockFn().resolvesTo(mockAnomalies)
+      indexer.getAnomalies = mockCalculateAnomalies
+
+      const from = MIN.toNumber()
+      const to = NOW
+
+      const result = await indexer.update(from, to.toNumber())
+
+      expect(mockCalculateAnomalies).toHaveBeenCalledWith(NOW.toStartOf('day'))
+
+      expect(mockAnomaliesRepository.addOrUpdateMany).toHaveBeenCalledWith(
+        mockAnomalies,
+      )
+
+      expect(result).toEqual(NOW.toStartOf('day').toNumber())
+    })
+  })
+
```

Comment:
aren't we missing test that just updates without adjusting?

Reply:
Added